src/java/org/apache/cassandra/index/sai/disk/hnsw/CassandraOnHeapHnsw.java

```diff
@@ -273,10 +275,18 @@ public PriorityQueue<T> search(float[] queryVector, int limit, Bits toAccept, in
         {
             throw new RuntimeException(e);
         }
-        PriorityQueue<T> keyQueue = new PriorityQueue<>();
+        PriorityQueue<PrimaryKey> keyQueue = new PriorityQueue<>();
```

Comment:
Not related to this, but does this need to be a `PriorityQueue`? The only use of this call passes the result to a `ReorderingRangeIterator` that could sort the keys before pulling them. My understanding is that this would be slightly more performant.

Reply:
PQ is better if you only read some of the entries.  I believe that was the idea in RRI but I'm not sure how often that's actually the case.